### DISCUSSION TITLE
Using trash in Makefile

### DISCUSSION THREAD

**Discussion by metropol** [argmaxinc/WhisperKit#47] [General]
`trash`, used in the following make rule, is not part of a default macOS setup.

```
clean-package-caches:
	@trash ~/Library/Caches/org.swift.swiftpm/repositories
	@trash ~/Library/Developer/Xcode/DerivedData
```

I see three options to address this:

1. use `rm` instead, and delete immediately
2. use `mv`, and move the files to the user's trash in `~/.Trash` (only works properly if the files are in the local disk; for external hard drives trashes are at `/Volumes/NAME_OF_EXTERNAL/.Trashes/USER_ID/`, and to handle these cases probably better go with option 3)
3. install `trash` using Homebrew in the `setup` rule.

**Comment by ZachNagengast** [comment]
Good callout, we're using trash here to avoid any permanent data loss, since this function is generally a last resort when xcode is having major swift package management issues. I'll create an issue to add [`brew install trash`](https://formulae.brew.sh/formula/trash) to the setup script.

**Reply by ZachNagengast** [reply]
Resolved with https://github.com/argmaxinc/WhisperKit/commit/7dbece081a22ccf9074819ca144778858b0f1758
